src/bootstrap/src/utils/exec.rs

```diff
@@ -84,10 +84,57 @@ pub struct CommandProfiler {
 }
 
 impl CommandProfiler {
-    pub fn record(&self, key: CommandCacheKey, exec_trace: ExecutionTrace) {
+    pub fn record_execution(&self, key: CommandCacheKey, start_time: Instant) {
         let mut stats = self.stats.lock().unwrap();
         let entry = stats.entry(key).or_default();
-        entry.exec_traces.push(exec_trace);
+        entry.traces.push(ExecutionTrace::Executed {
+            timestamp: start_time,
+            duration: start_time.elapsed(),
+        });
+    }
+
+    pub fn record_cache_hit(&self, key: CommandCacheKey) {
+        let mut stats = self.stats.lock().unwrap();
+        let entry = stats.entry(key).or_default();
+        entry.traces.push(ExecutionTrace::CacheHit { timestamp: Instant::now() });
+    }
+
+    pub fn report_summary(&self) {
+        let stats = self.stats.lock().unwrap();
+
+        for (key, profile) in stats.iter() {
```

Comment:
Could we write it into a file, e.g. `bootstrap-profile-<process PID>.txt`? It will be annoying to separate the output from the normal output of bootstrap otherwise.